lib/ansible/modules/subversion.py

```diff
@@ -295,7 +295,7 @@ def main():
 
     # We screenscrape a huge amount of svn commands so use C locale anytime we
     # call run_command()
-    module.run_command_environ_update = dict(LANG='C', LC_MESSAGES='C')
+    module.run_command_environ_update = dict(LANG='C', LC_MESSAGES='C', LC_ALL='en_US.UTF-8')
```

Comment:
C/POSIX is always available en_US is not guaranteed

Reply:
Ok. And the reason I needed use a UTF-8 locale is because the paths are not being converted to bytes, as we discussed elsewhere. I'll work on that on Monday.

Reply:
We could potentially attempt `C.UTF-8`, but that may not exist either.  It looks like fedora/rhel may not have always had it: https://bugzilla.redhat.com/show_bug.cgi?id=902094

Some research may need to be done.

Reply:
that is a bit newer, so in the future it will become the default/failover, but plenty of 'old' installs out there so we cannot rely on it yet.

Reply:
After converting the repo path to bytes, I am still getting the same error from `subversion` when the repo path contains non-ASCII characters if `LC_ALL=C`. The error below from subversion appears when attempting to checkout to `/tmp/tésting/xml-crimson`.

```
svn: E000022: Can't convert string from native encoding to 'UTF-8':
svn: E000022: /tmp/testing/xml-crimson
```

From reading the [documentation on how subversion handles localization](http://svnbook.red-bean.com/en/1.2/svn.advanced.l10n.html) I believe the issue is that subversion does not like that the current system locale does not support UTF-8.

> Errors like this typically occur when the Subversion client has received a UTF-8 string from the repository, but not all of the characters in that string can be represented using the encoding of the current locale. 

That makes me think that we won't be able to normalize the messages _and_ use paths that contain non-ASCII characters. 😞

Reply:
weee!

Reply:
or people just need to roll out their i18n version of subversion module to deal with messages in their native lang+encoding

Reply:
I looked at adjusting the regexp that searches for revisions and managed to get it working for all the locales on my system. But that seems like a pretty inaccurate fix. And I would need to do it for other regexp searches to make it complete...

Reply:
I talked to @mattclay about how best to test this and that ended up being a whole other big topic. Currently we don't have any tests running Ansible under a non-English locale. So the test would need to install the locale and configure it to make sure it's available (as @bcoca mentioned earlier).

I looked into using `C.UTF-8` and it's also pretty complicated, particularly for BSD and macOS. [PEP-538](https://www.python.org/dev/peps/pep-0538/) and [PEP-540](https://www.python.org/dev/peps/pep-0540/) provide some insight into how Python handled this.